src/util/generate/generate-project.ts

```diff
@@ -0,0 +1,66 @@
+import { relative } from 'path'
+import wait from '../output/wait'
+import { Output } from '../output'
+import { detectFromManifests } from './detect-from-manifests';
+import { outputFile, choose } from './helpers'
+import { internal } from './metadata'
+import { dirMap, generateDirMap } from './generate-dir-map'
+import { detectFromExtensions } from './detect-from-extensions';
+
+export type Build = { src: string, use: string }
+type nowJson = {
+  version: number,
+  name?: string,
+  builds: Array<Build>
+}
+type Project = { config: nowJson, ignore: string[] }
+
+async function processDir(root: string, map: dirMap, deepCapture: string[] = []): Promise<Build[]> {
+  const rel = relative(root, map.absolute)
+  
+  const manifestBuilds = await detectFromManifests(map.manifests, map.absolute, rel, { choose, outputFile })
+  if (manifestBuilds) return manifestBuilds
+
+  // Check extension groups
+  const { builds, capture } = await detectFromExtensions(map, deepCapture, rel, { choose })
+
+  // Check directories
+  for (let name in map.dir) {
+    if (Object.prototype.hasOwnProperty.call(map.dir, name)) {
+      builds.push(...await processDir(root, map.dir[name], capture.concat(deepCapture)))
+    }
+  }
+
+  return builds
+}
+
+export async function generateProject(dir: string, output: Output): Promise<Project> {
+  const stopSpinner = wait(`Loooking for code to build...`)
+  const ignore: {[key: string]: true} = {}
```

Comment:
Why are you using an object for ignoring instead of an array? Does anything use `ignore['path'] = false`?

Reply:
It's an object so there's no duplicates

Reply:
If that's the case, then I suggest using `new Set<boolean>()`